Sources/Build/ManifestBuilder.swift

```diff
@@ -149,6 +149,78 @@ extension LLBuildManifestBuilder {
     private func createSwiftCompileCommand(
         _ target: SwiftTargetBuildDescription
     ) {
+        // Inputs.
+        let inputs = computeSwiftCompileCmdInputs(target)
+
+        // Outputs.
+        let objectNodes = target.objects.map(Node.file)
+        let moduleNode = Node.file(target.moduleOutputPath)
+        let cmdOutputs = objectNodes + [moduleNode]
+
+        if buildParameters.emitSwiftModuleSeparately {
+            addSwiftCmdsEmitSwiftModuleSeparately(target, inputs: inputs, objectNodes: objectNodes, moduleNode: moduleNode)
+        } else {
+            addCmdWithBuiltinSwiftTool(target, inputs: inputs, cmdOutputs: cmdOutputs)
+        }
+
+        addTargetCmd(target, cmdOutputs: cmdOutputs)
+        addModuleWrapCmd(target)
+    }
+
+    private func addSwiftCmdsEmitSwiftModuleSeparately(
+        _ target: SwiftTargetBuildDescription,
+        inputs: [Node],
+        objectNodes: [Node],
+        moduleNode: Node
+    ) {
+        // FIXME: We need to ingest the emitted dependencies.
+
+        manifest.addShellCmd(
+            name: target.moduleOutputPath.pathString,
+            description: "Emitting module for \(target.target.name)",
+            inputs: inputs,
+            outputs: [moduleNode],
+            args: target.emitModuleCommandLine()
+        )
+
+        let cmdName = target.target.getCommandName(config: buildConfig)
+        manifest.addShellCmd(
+            name: cmdName,
+            description: "Compiling module \(target.target.name)",
+            inputs: inputs,
+            outputs: objectNodes,
+            args: target.emitObjectsCommandLine()
+        )
+    }
+
+    private func addCmdWithBuiltinSwiftTool(
+        _ target: SwiftTargetBuildDescription,
+        inputs: [Node],
+        cmdOutputs: [Node]
+    ) {
+        let isLibrary = target.target.type == .library || target.target.type == .test
```

Comment:
Can we make this a query on `target.target.type`? (Can you tell I'm not a fan of checking specific cases on enums?)